src/libtrx/game/ui/elements/fps_counter.c

```diff
@@ -0,0 +1,43 @@
+#include "game/ui/elements/fps_counter.h"
+
+#include "game/clock.h"
+#include "game/ui/elements/label.h"
+#include "memory.h"
+
+#include <stdio.h>
+
+typedef struct UI_FPS_COUNTER_STATE {
+    int32_t drawn_frames;
+    int32_t fps_counter;
+    CLOCK_TIMER timer;
+} UI_FPS_COUNTER_STATE;
+
+UI_FPS_COUNTER_STATE *UI_FPSCounter_Init(void)
+{
+    UI_FPS_COUNTER_STATE *const s = Memory_Alloc(sizeof(UI_FPS_COUNTER_STATE));
+    s->timer.type = CLOCK_TIMER_REAL;
+    return s;
+}
+
+void UI_FPSCounter_Free(UI_FPS_COUNTER_STATE *const s)
+{
+    if (s != nullptr) {
+        Memory_Free(s);
+    }
+}
+
+void UI_FPSCounter(UI_FPS_COUNTER_STATE *const s)
+{
+    char text[20];
+    if (s->fps_counter > 0) {
+        sprintf(text, "%d FPS", s->fps_counter);
+    } else {
+        sprintf(text, "? FPS");
```

Comment:
I can't seem to get this particular line to kick in. If I initially make the counter visible, it shows a low value (like 1 or 26 e.g.) before settling on 60.

Reply:
Thanks – it was just defensive programming on my part. Simplified the check, I think it's fine.